seed/seed20.go

```diff
@@ -353,6 +425,91 @@ func (e *noSnapDeclarationError) Error() string {
 	return fmt.Sprintf("cannot find snap-declaration for snap name: %s", e.snapRef.SnapName())
 }
 
+type errorComponentNotInSeed struct {
+	error
+}
+
+func (s *seed20) lookupVerifiedComponent(cref naming.ComponentRef, snapRev snap.Revision, snapID, snapProvenance string, snapsDir string, handler ContainerHandler, tm timings.Measurer) (Component, error) {
+	snapName := cref.SnapName
+	compName := cref.ComponentName
+
+	resKey := resourceKey{snapID: snapID, name: compName}
+	resRev, ok := s.resRevByResKey[resKey]
+	if !ok {
+		// No assertions might be ok if the component is optional, the
+		// caller should check for this error type in that case.
+		return Component{}, errorComponentNotInSeed{
+			fmt.Errorf("resource revision assertion not found for %s", compName)}
+	}
+	resPair, ok := s.resPairByResKey[resKey]
+	if !ok {
+		return Component{}, errorComponentNotInSeed{
+			fmt.Errorf("resource pair assertion not found for %s", compName)}
+	}
+
+	compPath := filepath.Join(s.systemDir, snapsDir,
+		fmt.Sprintf("%s_%d.comp", cref.String(), resRev.ResourceRevision()))
+	_, err := os.Stat(compPath)
+	if err != nil {
+		// error should be of type *PathError
+		return Component{}, errorComponentNotInSeed{err}
+	}
+
+	// Checks
+
+	// Note that the check for matching revisions in resource-revision /
+	// resource-pair is already done in LoadAssertions
+	if resPair.SnapRevision() != snapRev.N {
+		return Component{}, fmt.Errorf(
+			"resource %s pair revision does not match snap revision: %d != %d",
+			compName, resPair.SnapRevision(), snapRev.N)
+	}
+
+	if resRev.Provenance() != snapProvenance {
+		return Component{}, fmt.Errorf(
+			"resource revision provenance for %s does not match snap provenance: %s != %s",
+			compName, resRev.Provenance(), snapProvenance)
+	}
+	if resPair.Provenance() != snapProvenance {
+		return Component{}, fmt.Errorf(
+			"resource pair provenance for %s does not match snap provenance: %s != %s",
+			compName, resPair.Provenance(), snapProvenance)
+	}
```

Comment:
we probabl want some tests that fail some of this checks, see the error case tests for lookupVerifiedRevision for some ideas

Reply:
Added now